HPXMLtoOpenStudio/resources/meta_measure.rb

```diff
@@ -624,3 +624,38 @@ def is_integer?
     return true
   end
 end
+
+# TODO
+def convertArgumentValues(arguments_model, args, delete_auto = true)
+  args.each do |name, value|
+    if delete_auto && value == Constants::Auto
+      args.delete(name)
+      next
+    end
+
+    arg = arguments_model.find { |a| name == a.name.to_sym }
+    type = arg.type
+
+    if type == 'Choice'.to_OSArgumentType
+      choices = arg.choiceValues
+      if choices.include?(Constants::Auto)
+        if value.downcase.to_s == 'true'
+          args[name] = true
+        elsif value.downcase.to_s == 'false'
+          args[name] = false
+        end
+      end
+    elsif type == 'String'.to_OSArgumentType
+      begin
+        value = Float(value)
+        if value % 1 == 0
+          args[name] = Integer(value)
+        else
+          args[name] = value
+        end
+      rescue
+      end
```

Comment:
This is a bad idea. You should only be converting to integer/float for arguments that we've declared as integer/float, not simply converting anytime it's possible. I assume this is the reason that you then had to change a bunch of code so that it converts back to string.

You'll need to refactor the code so that you have access to the original desired argument type and use it here.

Reply:
Also, should this really be in meta_measure.rb? Do you anticipate using this anywhere else than the BuildResHPXML measure?

Reply:
I put it here because I found that ResStock needs to use it (with `delete_auto = false`).